compiler/rustc_passes/src/dead.rs

```diff
@@ -405,6 +408,13 @@ impl<'tcx> MarkSymbolVisitor<'tcx> {
                     intravisit::walk_item(self, item)
                 }
                 hir::ItemKind::ForeignMod { .. } => {}
+                hir::ItemKind::Fn(..) => {
+                    // check `T::Ty` in the types of inputs and output
+                    // the result of type_of maybe different from the fn sig,
+                    // so we also check the fn sig
+                    self.visit_middle_fn_sig(item.owner_id.def_id);
```

Comment:
Why do we need lowered types? Can't we use HIR?

Reply:
I don't get it. typeck results should give you the correct resolution, and not Res::Err. Or is there a special case I'm missing?